tests/test_js_signing.py

```diff
@@ -495,3 +495,33 @@ def test_contentScript_static_values(self):
             tab.attach({ contentScript: 'doc.createElement("script")' });
         """)
         self.assert_failed(with_warnings=[warning])
+
+
+class TestObserverEvents(TestCase, RegexTestCase):
+    """Tests that code related to observer events trigger warnings."""
+
+    def test_on_newtab_changed(self):
+        """
+        Tests that adding an observer to the `newtab-url-changed` event triggers
+        a high signing severity warning.
+        """
+        patterns = (
+            'Services.obs.addObserver(NewTabObserver, "newtab-url-changed", false);',
+            (
+              'var foo = Cc["foo"].getService(Ci.nsIObserverService);'
+              'foo.addObserver(NewTabObserver, "newtab-url-changed", false);'
```

Comment:
which line is the one being caught? It seems you're using two different ways to add the observer, but in the same script, don't you want to have one test for each?